### PR TITLE
Homework Node.js Week#1 Implemented

### PR DESCRIPTION
**Pull request by DiegoRuiz454**
app.js tested every file run without errors and with the correct results on the http://localhost:3000 and the file warmup can be tested on the server.

### CODE REVIEW

File: nodejs/week1/src/backend/app.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+app.get("/", async (request, response) => {
+    response.send("Meals Sharing Web App");
+}); 
+
+
+//----------------Respond with the json for all the meals.---------------------
+
+app.get("/meals", async (request, response) => {
+    
+    response.send(module.exports = {meals,reviews});
```
Comment (markuskoehler360) [inline nodejs/week1/src/backend/app.js]: are you passing all the tests? i think you are missing the part, where you are adding the relevant reviews to the meal objects... read line 172 in homework.md

File: nodejs/week1/src/backend/app.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}); 
+
+//-----------Respond with the json for all the meals that are cheap ---------------------
+
+app.get("/cheap-meals", async (request, response) => {
+    
+    const cheap = meals.filter(meals => meals.price <= 50);
+    response.send(module.exports = cheap);
+});
```
Comment (markuskoehler360) [inline nodejs/week1/src/backend/app.js:30]: you forgot to modularize the routes (line 183 in homework.md)

File: nodejs/week1/src/backend/app.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+const warmup = require("./data/warmup.js");
+
+//-----the result of the of the circle measurements in the server----------------
+
+console.log(warmup); 
+
+//----------Meals sharing Web ---------------------------
+
+app.get("/", async (request, response) => {
+    response.send("Meals Sharing Web App");
```
Comment (markuskoehler360) [inline nodejs/week1/src/backend/app.js]: here is a small typo, that leads to a failed test! i know it has nothing to do with your code, but try to run the test suite one more time, to see what i mean ;)

File: nodejs/week1/src/backend/data/warmup.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    getCircumference ()
+    {
+        return 2 * Math.PI * this.radius;
+    };
+  //--------------- get Area ---------------------------
+
+     getArea ()
+     {
+         return Math.PI * this.radius * this.radius ;
+     };}
```
Comment (markuskoehler360) [inline nodejs/week1/src/backend/data/warmup.js]: here you can improve the code formatting a little bit, to make it more readible, are you using VS Code? there is a nice extension called 'prettier' that takes care of code formatting for you 👍

Comment (DiegoRuiz454) [reply nodejs/week1/src/backend/data/warmup.js]: thanks for the feedback i will fix the errors i have in my code.

File: nodejs/week1/src/backend/app.js (hunk 1 above)
Comment (DiegoRuiz454) [reply nodejs/week1/src/backend/app.js]: Is true i forgot this point in the task. I will do

**Pull request merged**